diff --git a/dist/parser/typescript.js b/dist/parser/typescript.js
index 4165e215214cae0e6a550d38284d0181a6ab3448..1653b428517a7574f73a0dec6c6bb3962b17d739 100644
--- a/dist/parser/typescript.js
+++ b/dist/parser/typescript.js
@@ -17,7 +17,7 @@ async function parseTypescript(filename) {
     // ['decorators', { decoratorsBeforeExport: true }],
     'decorators-legacy', 'doExpressions', 'dynamicImport', 'exportDefaultFrom', 'exportNamespaceFrom', 'functionBind', 'functionSent', 'importMeta', 'logicalAssignment', 'nullishCoalescingOperator', 'numericSeparator', 'objectRestSpread', 'optionalCatchBinding', 'optionalChaining', ['pipelineOperator', {
       proposal: 'minimal'
-    }], 'throwExpressions', 'importAssertions']
+    }], 'throwExpressions', 'importAssertions', 'explicitResourceManagement']
   });
 }
 module.exports = exports.default;
\ No newline at end of file
