diff --git a/esm/useThrottle.d.ts b/esm/useThrottle.d.ts
index bb20b9649ddaef1015e027ef14ce9a557eabc1b3..9f9173a7d7308d9c573947305976d88c28f0c984 100644
--- a/esm/useThrottle.d.ts
+++ b/esm/useThrottle.d.ts
@@ -1,2 +1,2 @@
-declare const useThrottle: <T>(value: T, ms?: number) => T;
+declare const useThrottle: <T>(value: T | (() => T), ms?: number) => T;
 export default useThrottle;
diff --git a/lib/useThrottle.d.ts b/lib/useThrottle.d.ts
index bb20b9649ddaef1015e027ef14ce9a557eabc1b3..9f9173a7d7308d9c573947305976d88c28f0c984 100644
--- a/lib/useThrottle.d.ts
+++ b/lib/useThrottle.d.ts
@@ -1,2 +1,2 @@
-declare const useThrottle: <T>(value: T, ms?: number) => T;
+declare const useThrottle: <T>(value: T | (() => T), ms?: number) => T;
 export default useThrottle;
