diff --git a/dist/esm/client/preview/render.js b/dist/esm/client/preview/render.js
index 4a388101c8db468e1a4c9805c5b5a6a364655487..43a5fda1a29a58df0ee07c4c999d02d282589665 100644
--- a/dist/esm/client/preview/render.js
+++ b/dist/esm/client/preview/render.js
@@ -53,14 +53,16 @@ export var render = function render(args, context) {
 };

 var renderElement = /*#__PURE__*/function () {
-  var _ref = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee(node, el) {
+  var _ref = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee(node, el, storyContext) {
     return regeneratorRuntime.wrap(function _callee$(_context) {
       while (1) {
         switch (_context.prev = _context.next) {
           case 0:
             return _context.abrupt("return", new Promise(function (resolve) {
               ReactDOM.render(node, el, function () {
-                return resolve(null);
+                // access the parameters in the render callback to allow decorators to set storyReady
+                var maybeSignal = storyContext.parameters?.storyReady;
+                maybeSignal ? maybeSignal.then(() => resolve(null)) : resolve(null);
               });
             }));

@@ -164,7 +166,7 @@ function _renderToDOM() {
             }

             _context2.next = 7;
-            return renderElement(element, domElement);
+            return renderElement(element, domElement, storyContext);

           case 7:
           case "end":
