diff --git a/dist/esm/client/preview/render.js b/dist/esm/client/preview/render.js
index 5bf30e1d97021f22c529384316e215df1537f302..43567af6f9b47d550f59898caa21e75a9b4121bc 100644
--- a/dist/esm/client/preview/render.js
+++ b/dist/esm/client/preview/render.js
@@ -57,7 +57,7 @@ export var render = function render(args, context) {
 };

 var renderElement = /*#__PURE__*/function () {
-  var _ref = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee(node, el) {
+  var _ref = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee(node, el, storyContext) {
     var root;
     return regeneratorRuntime.wrap(function _callee$(_context) {
       while (1) {
@@ -76,7 +76,9 @@ var renderElement = /*#__PURE__*/function () {
                 }, 0);
               } else {
                 ReactDOM.render(node, el, function () {
-                  return resolve(null);
+		  // access the parameters in the render callback to allow decorators to set storyReady
+                  var maybeSignal = storyContext.parameters?.storyReady;
+                  maybeSignal ? maybeSignal.then(() => resolve(null)) : resolve(null);
                 });
               }
             }));
@@ -242,7 +244,7 @@ function _renderToDOM() {
             }

             _context3.next = 7;
-            return renderElement(element, domElement);
+            return renderElement(element, domElement, storyContext);

           case 7:
           case "end":
