diff --git a/dist/esm/client/preview/render.js b/dist/esm/client/preview/render.js
index c2f6c314c82f96d1b5be90cd957e0f1926e8bd28..0ff99c0918251a9fd58b80bfbfe5e225bfd807ff 100644
--- a/dist/esm/client/preview/render.js
+++ b/dist/esm/client/preview/render.js
@@ -70,7 +70,7 @@ var WithCallback = function WithCallback(_ref) {
 };

 var renderElement = /*#__PURE__*/function () {
-  var _ref2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee(node, el) {
+  var _ref2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee(node, el, storyContext) {
     var root;
     return regeneratorRuntime.wrap(function _callee$(_context) {
       while (1) {
@@ -85,12 +85,16 @@ var renderElement = /*#__PURE__*/function () {
               if (root) {
                 root.render( /*#__PURE__*/React.createElement(WithCallback, {
                   callback: function callback() {
-                    return resolve(null);
+                    // access the parameters in the render callback to allow decorators to set storyReady
+                    var maybeSignal = storyContext.parameters?.storyReady;
+                    maybeSignal ? maybeSignal.then(() => resolve(null)) : resolve(null);
                   }
                 }, node));
               } else {
                 ReactDOM.render(node, el, function () {
-                  return resolve(null);
+                  // access the parameters in the render callback to allow decorators to set storyReady
+                  var maybeSignal = storyContext.parameters?.storyReady;
+                  maybeSignal ? maybeSignal.then(() => resolve(null)) : resolve(null);
                 });
               }
             }));
@@ -256,7 +260,7 @@ function _renderToDOM() {
             }

             _context3.next = 7;
-            return renderElement(element, domElement);
+            return renderElement(element, domElement, storyContext);

           case 7:
           case "end":
